docs/main.typ

```diff
@@ -0,0 +1,233 @@
+          \
+          \
+          —  Plutarch, Life of Theseus 23.1
+        ]
+      )
+      \
+      \
+      \
+      \
+      \
+      \
+      \
+      \
+      \
+      \
+      \
+      \
+      \
+      \
+      #box(
+        width: 94mm,
+        [
+I would like to thank Christian Gram Kalhauge \<chrg\@dtu.dk\> for proposing the idea for this project and for his valuable external supervision and guidance.])
+
+
+
+    ]
+  ],
+  bibliography: bibliography("refs.bib"),
+  figure-index: (enabled: true),
+  table-index: (enabled: true),
+  listing-index: (enabled: true),
+)
+
+= Introduction
+
+
+
+RQ1 *Is the compression distance a more representative metric for quantifying the complexity than lines of code changed?*
+
+RQ2 *Can the compression distance tell us about the complexity of different categories of commits?*
+
+RQ3 *Does the compression distance suffer from the same limitations as lines of code changed in quantifying the contributions of developers?*
+
+= Introduction to Similarity Distance Metrics
+
+== Information distance
+
+In the field of information theory, the concept of _information distance_ is used to quantify the similarity between two objects (TODO:REF). This is done by measuring the amount of information needed to transform one object into another. The most common way to measure this distance is by using a _distance metric_, which is a function that quantifies the difference between two objects.
+
+While there exists no perfect such metric, we can approximate the information distance between two objects using various practical techniques.
+
+== Lines of Code Changed as a measure of information distance
+
+_Lines of code_ (LoC) @Sourceli1:online is one of the most widely used distance metrics in the software industry @nguyen2007sloc. It is often used as a measure of software complexity, maintainability, and productivity.
+
+
+Since many projects employ version control systems, such as Git (TODO: Ref to Git), for keeping track of changes, we can we can track the _Lines of code changed_ (LoCC) over time using diffing algorithms. The LoCC metric is typically defined as the number of lines added and removed in a commit. (TODO: source?).
+This provides a measure of the information distance between revisions of a software system. Git includes this functionality by default (TODO: Ref numstat).
+
+This is a commonly used technique used to detect activity in software systems over time @goeminne2013analyzing. It can be used to assess team velocity, developer productivity and more. These metrics can be automatically obtained via version control systems using tools like Git Truck@hojelse2022git. LoCC is a useful metric for quantifying contributions or regions of interest in software systems over time and tools like Git Truck have proven the effectiveness in the analysis of software evolution @lungu2022can, @neyem2025exploring.
```

Comment:
Er det måske lidt en stretch at sige at Git Truck har "proven" det?

Reply:
Well der er to papers der siger det 😅😅